### PR TITLE
[Bugfix][Spec Decode] Warm spec-sized logits all-gather during init

### PR DESCRIPTION
**Pull request by haosdent**
## Purpose

Fixes DeepSeek/GLM-5.2 MTP speculative decoding hanging at startup (#48568).

Warmup runs `compute_logits`' vocab-parallel all-gather at one row per request, but spec decode verifies `max_num_reqs*(1+num_speculative_tokens)` rows. On ROCm/RCCL the larger message connects extra channels lazily on first use, so when that first use is a live decode step the connect fails (`no transport for peer on channel N`) and the engine hangs. Warm `compute_logits` at the spec-decode row count during init.

## Test Plan

Serve GLM-5.2-FP8 with MTP speculative decoding on 8x MI300X (gfx942):

```
vllm serve zai-org/GLM-5.2-FP8 --tensor-parallel-size 8 \
  --kv-cache-dtype fp8_e4m3 --linear-backend aiter --moe-backend aiter \
  --speculative-config '{"method":"mtp","num_speculative_tokens":1}'
```

## Test Result

Server starts and serves normally with MTP speculative decoding, producing coherent completions. Without this change the engine hangs in the logits all-gather on the first speculative-decode step.

### CODE REVIEW

**Comment by haosdent** [comment]
@AndreasKaratzas can you help to take a look, this is to fix MTP issues when I run it in AMD Mi300x

**Comment by cjackal** [comment]
Seems like item 1 has been covered in #48036

**Comment by haosdent** [comment]
@vllmellm I have rebase on the latest main

**Comment by haosdent** [comment]
Thx @cjackal , have rebased.
